sql/opt/exec/execbuilder: TestExecBuild failed

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4758383&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4758383&tab=artifacts#/) on master @ [fa0291175ecb983301d980247ad2ec6c7b233660](https://github.com/cockroachdb/cockroach/commits/fa0291175ecb983301d980247ad2ec6c7b233660):


```
[08:13:27] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1817/execroot/cockroach/bazel-out/k8-dbg/bin/pkg/sql/opt/exec/execbuilder/execbuilder_test_/execbuilder_test.runfiles/cockroach/pkg/sql/opt/exec/execbuilder/testdata/show_trace_nonmetamorphic with config local: 60 tests, 1 failures
[08:13:27] rng seed: -1466258873313100092

            --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic/system_table_lookup (1.47s)
=== RUN   TestExecBuild/local/show_trace_nonmetamorphic
=== PAUSE TestExecBuild/local/show_trace_nonmetamorphic
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic
=== CONT  TestExecBuild/local/show_trace_nonmetamorphic
    logic.go:2976: let $trace_query = 
          SELECT operation,
                 regexp_replace(
              regexp_replace(
                regexp_replace(
                  regexp_replace(
                    regexp_replace(
                      regexp_replace(
                        message,
                        'create_as_of_time:<[^>]+>',
                        'create_as_of_time:<...>'
                      ),
                      e'created_at_nanos:[1-9]\\d*',
                      'created_at_nanos:...',
                      'g'
                    ),
                    'mutationJobs:<[^>]*>',
                    'mutationJobs:<...>'
                  ),
                  e'drop_time:[1-9]\\d*',
                  'drop_time:...'
                ),
                'declarative_schema_changer_state:<[^>]*>',
                'declarative_schema_changer_state:<...>'
              ),
              'metadata:<[^>]*>',
              'metadata:<...>'
                 ) AS message
            FROM [SHOW KV TRACE FOR SESSION]
           WHERE message NOT LIKE '%Z/%'
                 AND operation NOT LIKE 'kv.DistSender: sending partial batch%'
                 AND message NOT SIMILAR TO '%(PushTxn|ResolveIntent|SystemConfigSpan)%'
                 AND tag NOT LIKE '%intExec=%'
                 AND tag NOT LIKE '%scExec%'
                 AND tag NOT LIKE '%IndexBackfiller%'
                 AND operation != 'dist sender send';
        
[08:13:23] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1817/execroot/cockroach/bazel-out/k8-dbg/bin/pkg/sql/opt/exec/execbuilder/execbuilder_test_/execbuilder_test.runfiles/cockroach/pkg/sql/opt/exec/execbuilder/testdata/tpch_vec with config local: 39 tests, 0 failures
[08:13:23] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1817/execroot/cockroach/bazel-out/k8-dbg/bin/pkg/sql/opt/exec/execbuilder/execbuilder_test_/execbuilder_test.runfiles/cockroach/pkg/sql/opt/exec/execbuilder/testdata/srfs with config local: 22 tests, 0 failures
[08:13:23] rng seed: -2437789668618782543

        --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic (7.09s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-14635

--- a/pkg/sql/colfetcher/index_join.go
+++ b/pkg/sql/colfetcher/index_join.go
@@ -440,10 +440,19 @@ func (s *ColIndexJoin) GetCumulativeContentionTime() time.Duration {
 // we can remove this limit.
 var inputBatchSizeLimit = int64(util.ConstantWithMetamorphicTestRange(
 	"ColIndexJoin-batch-size",
-	4<<20, /* 4 MB */
-	1,     /* min */
-	4<<20, /* max */
+	productionIndexJoinBatchSize, /* defaultValue */
+	1,                            /* min */
+	productionIndexJoinBatchSize, /* max */
 ))
+
+const productionIndexJoinBatchSize = 4 << 20 /* 4MiB */
+
+func getIndexJoinBatchSize(forceProductionValue bool) int64 {
+	if forceProductionValue {
+		return productionIndexJoinBatchSize
+	}
+	return inputBatchSizeLimit
+}
 
 // NewColIndexJoin creates a new ColIndexJoin operator.
 //
@@ -528,7 +537,7 @@ func NewColIndexJoin(
 		usesStreamer:     useStreamer,
 		limitHintHelper:  execinfra.MakeLimitHintHelper(spec.LimitHint, post),
 	}
-	op.mem.inputBatchSizeLimit = inputBatchSizeLimit
+	op.mem.inputBatchSizeLimit = getIndexJoinBatchSize(flowCtx.EvalCtx.TestingKnobs.ForceProductionBatchSizes)
 	op.prepareMemLimit(inputTypes)
 	if useStreamer {
 		op.streamerInfo.budgetLimit = 3 * memoryLimit
@@ -537,7 +546,7 @@ func NewColIndexJoin(
 			return nil, errors.AssertionFailedf("diskMonitor is nil when ordering needs to be maintained")
 		}
 		op.streamerInfo.diskMonitor = diskMonitor
-		if memoryLimit < inputBatchSizeLimit {
+		if memoryLimit < op.mem.inputBatchSizeLimit {
 			// If we have a low workmem limit, then we want to reduce the input
 			// batch size limit.
 			//

